### PR TITLE
Use standard std::chrono based time for merge operation. MERGEOK

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR

### CODE REVIEW

File: storage/src/vespa/storage/distributor/operations/idealstate/mergeoperation.cpp [hunk 1]
```diff
@@ -184,28 +178,20 @@ MergeOperation::sourceOnlyCopyChangedDuringMerge(
         const BucketDatabase::Entry& currentState) const
 {
     assert(currentState.valid());
-    for (size_t i = 0; i < _mnodes.size(); ++i) {
-        const BucketCopy* copyBefore(_infoBefore.getNode(_mnodes[i].index));
+    for (auto mnode : _mnodes) {
```
Comment (vekterli) [inline storage/src/vespa/storage/distributor/operations/idealstate/mergeoperation.cpp:181]: Can this be `const auto&`?

File: storage/src/vespa/storage/distributor/operations/idealstate/mergeoperation.cpp [hunk 2]
```diff
@@ -220,25 +206,22 @@ MergeOperation::deleteSourceOnlyNodes(
 {
     assert(currentState.valid());
     std::vector<uint16_t> sourceOnlyNodes;
-    for (uint32_t i = 0; i < _mnodes.size(); ++i) {
-        const uint16_t nodeIndex = _mnodes[i].index;
+    for (auto & mnode : _mnodes) {
```
Comment (vekterli) [inline storage/src/vespa/storage/distributor/operations/idealstate/mergeoperation.cpp:209]: Consider `const`-ifying ref

File: storage/src/vespa/storage/distributor/operations/idealstate/mergeoperation.cpp [hunk 3]
```diff
@@ -33,11 +35,11 @@ MergeOperation::addIdealNodes(
         std::vector<MergeMetaData>& result)
 {
     // Add all ideal nodes first. These are never marked source-only.
-    for (uint32_t i = 0; i < idealNodes.size(); i++) {
+    for (unsigned short idealNode : idealNodes) {
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/vespa/storage/distributor/operations/idealstate/mergeoperation.cpp:38]: Fixed in #25921

File: storage/src/vespa/storage/distributor/operations/idealstate/mergeoperation.cpp [hunk 4]
```diff
@@ -50,21 +52,20 @@ MergeOperation::addIdealNodes(
...
-MergeOperation::addCopiesNotAlreadyAdded(
-        uint16_t redundancy,
-        const std::vector<MergeMetaData>& nodes,
-        std::vector<MergeMetaData>& result)
+MergeOperation::addCopiesNotAlreadyAdded(uint16_t redundancy,
+                                         const std::vector<MergeMetaData>& nodes,
+                                         std::vector<MergeMetaData>& result)
 {
-    for (uint32_t i = 0; i < nodes.size(); i++) {
+    for (auto node : nodes) {
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/vespa/storage/distributor/operations/idealstate/mergeoperation.cpp:59]: yes, #25921

File: storage/src/vespa/storage/distributor/operations/idealstate/mergeoperation.cpp [hunk 5]
```diff
@@ -123,31 +123,25 @@ MergeOperation::onStart(DistributorStripeMessageSender& sender)
     std::vector<std::unique_ptr<BucketCopy> > newCopies;
     std::vector<MergeMetaData> nodes;
 
-    for (uint32_t i = 0; i < getNodes().size(); ++i) {
-        const BucketCopy* copy = entry->getNode(getNodes()[i]);
+    for (unsigned short node : getNodes()) {
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/vespa/storage/distributor/operations/idealstate/mergeoperation.cpp:126]: Done. #25921

File: storage/src/vespa/storage/distributor/operations/idealstate/mergeoperation.cpp (hunk 1 above)
Comment (baldersheim) [reply storage/src/vespa/storage/distributor/operations/idealstate/mergeoperation.cpp:181]: Fixed #25921

File: storage/src/vespa/storage/distributor/operations/idealstate/mergeoperation.cpp (hunk 2 above)
Comment (baldersheim) [reply storage/src/vespa/storage/distributor/operations/idealstate/mergeoperation.cpp:209]: Fixed in #25921

**Pull request merged**